system_tests/bigtable_happybase.py

```diff
@@ -406,6 +407,217 @@ def test_rows_with_timestamp(self):
+        self.assertEqual(first_two, [(value2, ts2), (value1, ts1)])
+
+        # Limit to cells before the 2nd timestamp (exclusive).
+        first_cell = table.cells(ROW_KEY1, COL1, include_timestamp=True,
+                                 timestamp=ts2)
+        self.assertEqual(first_cell, [(value1, ts1)])
+
+
+class TestTable_scan(BaseTableTest):
+
+    def test_scan_when_empty(self):
+        scan_result = list(Config.TABLE.scan())
+        self.assertEqual(scan_result, [])
+
+    def test_scan_single_row(self):
+        table = Config.TABLE
+        value1 = 'value1'
+        value2 = 'value2'
+        row1_data = {COL1: value1, COL2: value2}
+
+        # Need to clean-up row1 after.
+        self.rows_to_delete.append(ROW_KEY1)
+        table.put(ROW_KEY1, row1_data)
+
+        scan_result = list(table.scan())
+        self.assertEqual(scan_result, [(ROW_KEY1, row1_data)])
+
+        scan_result_cols = list(table.scan(columns=[COL1]))
+        self.assertEqual(scan_result_cols, [(ROW_KEY1, {COL1: value1})])
+
+        scan_result_ts = list(table.scan(include_timestamp=True))
+        self.assertEqual(len(scan_result_ts), 1)
+        only_row = scan_result_ts[0]
+        self.assertEqual(only_row[0], ROW_KEY1)
+        row_values = only_row[1]
+        ts = row_values[COL1][1]
+        self.assertEqual(row_values, {COL1: (value1, ts), COL2: (value2, ts)})
+
+    def test_scan_filters(self):
+        table = Config.TABLE
+        value1 = 'value1'
+        value2 = 'value2'
+        value3 = 'value3'
+        value4 = 'value4'
+        value5 = 'value5'
+        value6 = 'value6'
+        row1_data = {COL1: value1, COL2: value2}
+        row2_data = {COL2: value3, COL3: value4}
+        row3_data = {COL3: value5, COL4: value6}
+
+        # Need to clean-up row1/2/3 after.
+        self.rows_to_delete.append(ROW_KEY1)
+        self.rows_to_delete.append(ROW_KEY2)
+        self.rows_to_delete.append(ROW_KEY3)
+        table.put(ROW_KEY1, row1_data)
+        table.put(ROW_KEY2, row2_data)
+        table.put(ROW_KEY3, row3_data)
+
+        # Basic scan (no filters)
+        scan_result = list(table.scan())
+        self.assertEqual(scan_result, [
+            (ROW_KEY1, row1_data),
+            (ROW_KEY2, row2_data),
+            (ROW_KEY3, row3_data),
+        ])
+
+        # Limit the size of the scan
+        scan_result = list(table.scan(limit=1))
+        self.assertEqual(scan_result, [
+            (ROW_KEY1, row1_data),
+        ])
+
+        # Scan with a row prefix.
+        prefix = ROW_KEY2[:-1]
+        self.assertEqual(prefix, ROW_KEY3[:-1])
+        scan_result_prefixed = list(table.scan(row_prefix=prefix))
+        self.assertEqual(scan_result_prefixed, [
+            (ROW_KEY2, row2_data),
+            (ROW_KEY3, row3_data),
+        ])
+
+        # Make sure our keys are sorted in order
+        row_keys = [ROW_KEY1, ROW_KEY2, ROW_KEY3]
+        self.assertEqual(row_keys, sorted(row_keys))
```

Comment:
Made a list of three literal strings, and then sorted them and compared the results.  That is the bit I don't get:  nothing in our code could cause that assertion to fail.